src/code_8099360.c

```diff
@@ -0,0 +1,349 @@
+#include "global.h"
+#include "file_system.h"
+#include "event_flag.h"
+#include "code_80A26CC.h"
+
+extern struct FileArchive gSystemFileArchive;
+extern const u8 gUnknown_811601C[];
+extern u8 gUnknown_20398B8;
+
+extern void sub_8003600(void);
+
+extern s32 sub_800388C(u16, u8 *, u8);
+
+extern void sub_80A4B38(void);
+extern void sub_80A4B54(void);
+
+extern void sub_8099690(u32);
+
+bool8 sub_8099360(u8 *dungeonId) {
+    s16 script_disc;
+    DungeonInfo* dungeonInfo;
+
+    script_disc = GetScriptVarValue(NULL, DUNGEON_ENTER);
+    if (script_disc == 0x52) {
+        script_disc = GetScriptVarValue(NULL, DUNGEON_ENTER_INDEX);
+        dungeonInfo = (DungeonInfo *)GetDungeonInfo_80A2608(script_disc);
+            *dungeonId = dungeonInfo->dungeonIndex;
+            return TRUE;
+    }
+    return FALSE;
+}
+
+u8 sub_8099394(volatile u8 *param) {
+    s16 script_disc;
+
+    script_disc = GetScriptVarValue(NULL, DUNGEON_ENTER);
+    if (script_disc == 0x50) {
+        script_disc = GetScriptVarValue(NULL, DUNGEON_ENTER_INDEX);
+        *(u8 *)param = script_disc;
```

Comment:
What's going on here, why volatile and then another cast? Probably unneeded, but if it is needed, it deserves comments

Reply:
I guess my reasoning is because in `src/post_office_guide2.c` in `DisplayMissionObjectives()` a volatile gets passed in, `sub_8099394` gets a `volatile u8 local_94` passed in, but everywhere else it's called it uses just a regular `u8*`, so this way it preserves the properties of both. I'm not sure why `DisplayMissionObjectives()` uses a volatile there, but I didn't want to disrupt the functionality of the function there. Removing the volatile and cast gives the following warning/error:

```src/post_office_guide2.c: In function `DisplayMissionObjectives':
src/post_office_guide2.c:116: warning: passing arg 1 of `sub_8099394' discards qualifiers from pointer target type```

Reply:
Function still matches without the volatile. Usually we use it for variables to force it on the stack. Might want to try removing volatile from the function declaration in post_office_guide2.c and see if it still matches.

https://decomp.me/scratch/6BPvc <- decomp.me link for sub_8099394

Reply:
https://github.com/pret/pmd-red/blob/9e4958462189b8dad07f32e1c01931cacf349687/src/post_office_guide2.c#L98C1-L118C45

Ah I actually got it! If in the code above we change line 116 to `sub_8099394((u8 *)&local_94);` it compiles and `make compare` is OK. This resolves the dependency of volatile in sub_8099394. I can push the patch to this in a moment